### PR TITLE
[Counterfactual] Add pay now pay later option to safe creation

### LINKED ISSUE #3210: [Counterfactual] Implement Safe creation design
## What is this about

The safe creation design needs to be adjusted to support counterfactual deployments

## Requirements

- [ ] Add a pay now, pay later option for counterfactual safe setups
- [ ] Ignore relaying for counterfactual safe setups

## Screenshots

Screenshot 2024-02-06 at 14 50 49

### PR DESCRIPTION
**Pull request by usame-algan**
## What it solves

Resolves #3210 

## How this PR fixes it

- Adds a Pay now / Pay later option to the safe creation flow ONLY for counterfactual safes

## How to test it

1. Create a new 1/1 safe
2. Observe a Pay now / Pay later option on the review screen
3. Choose Pay later and create
4. You should immediately see the dashboard. Remember the safe address
5. Go through the safe creation flow again
6. Choose Pay later and either choose relayer or pay via wallet
7. Observe the transaction is being executed and the same safe address is deployed
8. Open the safe creation flow again and choose a different setup e.g. 1/2
9. Observe no Pay now / Pay later option
10. The safe should be created as normal

## Screenshots

Counterfactual without relaying:
Screenshot 2024-02-10 at 11 54 48

Counterfactual with relaying:
Screenshot 2024-02-10 at 11 55 12

No counterfactual with relaying:
Screenshot 2024-02-10 at 11 55 31

No counterfactual without relaying:
Screenshot 2024-02-10 at 11 55 50

## Checklist
* [ ] I've tested the branch on mobile 📱
* [ ] I've documented how it affects the analytics (if at all) 📊
* [ ] I've written a unit/e2e test for it (if applicable) 🧑‍💻

### CODE REVIEW

File: src/components/new-safe/create/steps/ReviewStep/index.tsx [hunk 1]
```diff
@@ -226,7 +228,16 @@ const ReviewStep = ({ data, onSubmit, onBack, setStep }: StepRenderProps<NewSafe
         </Grid>
       </Box>
 
-      {!isCounterfactual && (
+      {isCounterfactual && (
```
Comment (schmanu) [inline src/components/new-safe/create/steps/ReviewStep/index.tsx:231]: Should we add tests for this component behavior?

**Review by schmanu — APPROVED** [review]
Just a comment about testing. Otherwise it looks good :)

**Comment by francovenica** [comment]
LGTM

The Option to pay later only shows for safes 1/1 
After creating a CF safe, paying for the next one deploys the current CF. (even if you clear your LS)

**Comment by usame-algan** [comment]
Thanks @francovenica!

I've made some minor design updates according to [Figma](https://www.figma.com/file/fUOaMHVOhsIylBjxiL5uEn/Counterfactual-deployment?node-id=3832%3A9533&mode=dev) and updated the screenshots in this PR description in case you want to check again.

**Pull request merged**